### PR TITLE
Neural network unit tests and general architecture classes

### PR DESCRIPTION
**Pull request by lgraesser**

### CODE REVIEW

File: bin/setup_macOS [hunk 1]
```diff
@@ -55,6 +55,7 @@ echo "--- Installing npm modules for Lab interface ---"
 if [ -d ./node_modules ]; then
   echo "Npm modules are already installed"
 else
+  brew install yarn
```
Comment (kengz) [inline bin/setup_macOS]: this line already does it: https://github.com/kengz/Unity-Lab/blob/master/bin/setup_macOS#L51

**Comment by kengz** [comment]
nit-picking for style consistency :P 
1. test files shd be named `test_lorem_ipsum.py` (saw `datatest.py` and `nntest.py`
2. if possible, treat folders as module, and hence use singular. e.g. the `agent` module, `algorithm` module. This saves user from guessing "hmm `agent` or `agents`, what how to spell plural.

File: bin/setup_macOS (hunk 1 above)
Comment (lgraesser) [reply bin/setup_macOS]: Hmmm. That's odd. I got an install error when I ran the macOS script "yarn command not found". This line fixed it.

Comment (kengz) [reply bin/setup_macOS]: lul prolly sudo denied there. keep this then

**Comment by lgraesser** [comment]
The nn tests still need to be integrated into the automatic test suite

**Pull request merged**